content/tutorials/jwt-authentication.textile

```diff
@@ -136,96 +139,97 @@ This step shows you how to create the server code from scratch. If you get stuck
 
-9. To finish the web app you use "Express":https://www.npmjs.com/package/express to build a simple server to serve files on your localhost:
+const app = express();
 
-```[javascript]
-const express = require('express'),
-    app = express();
 app.use('/', express.static(__dirname))
 ```
 
 The Express server uses a middleware to serve the content, and uses @express.static(__dirname)@ to specify that any HTML files in the root folder are served directly.
 
-10. Create the @/auth@ route in this server. This route is responsible for creating a JWT and sending this token back to the client:
+8. Create the @/auth@ route in this server. This route is responsible for creating a JWT and sending this token back to the client. Normally we would validate the User before issuing the JWT token but for simplicity we'll just use their input as the `clientId`.
+
+In the front-end HTML we call the authUrl with a querysting variable, `/auth?clientId={username}:{password}`, which we access with the Express Request object. The parameters of the GET query are exposed by the @Request.query@ collection. In our example the users input will be available as @req.query.clientId@
 
 ```[javascript]
-app.get('/auth', (req, res) => {
-    console.log('Sucessfully connected to the server auth endpoint')
-    jwt.sign(jwtPayload, keySecret, jwtOptions, (err, tokenId) => {
-        console.log('JSON Web Token signed by auth server')
-        if (err) {
-            console.trace()
-            return
-        }
-        res.header('Cache-Control', 'private, no-cache, no-store, must-revalidate')
-        res.setHeader('Content-Type', 'application/json')
-        console.log('Sending signed JWT token back to client')
-        res.send(JSON.stringify(tokenId));
-    })
-                                                                                                                                                                                                                                                
-})
+app.get("/auth", (req, res) => {
+  console.log("Sucessfully connected to the server auth endpoint");
+
+  const randomId = Math.random().toString(16).slice(-8);
+  const clientId = req.query.clientId || randomId;
+
+  const jwtPayload = {
+    "x-ably-capability": capability,
+    "x-ably-clientId": clientId,
+  };
+
+  jwt.sign(jwtPayload, keySecret, jwtOptions, (err, tokenId) => {
+    console.log("JSON Web Token signed by auth server");
+
+    if (err) return console.trace();
+
+    res.header("Cache-Control", "private, no-cache, no-store, must-revalidate");
+    res.setHeader("Content-Type", "application/json");
+
+    console.log("Sending signed JWT token back to client", tokenId);
+    res.send(JSON.stringify(tokenId));
+  });
+});
 ```
 
-The previous code is invoked when a client application invokes the @/auth@ route on your auth server. 
+The previous code is invoked when a client application invokes the @/auth@ route on your auth server.
```

Comment:
Suggest putting this _before_ the code sample after the introductory text that says what it does. "... This code is invoked when a client application makes a request to the @/auth@ route on your server."